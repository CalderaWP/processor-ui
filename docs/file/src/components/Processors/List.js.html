<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../../">
  <title data-ice="title">src/components/Processors/List.js | Caldera Processor UI</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
<meta name="description" content="Caldera Forms processor user interface generator and components."><meta property="og:type" content="website"><meta property="og:url" content="http://calderalabs.org/processor-ui"><meta property="og:site_name" content="Caldera Processor UI"><meta property="og:title" content="Caldera Processor UI"><meta property="og:image" content="https://calderaforms.com/wp-content/themes/caldera-forms-dot-com/img/caldera-forms-logo.svg"><meta property="og:description" content="Caldera Forms processor user interface generator and components."><meta property="og:author" content="http://calderalabs.org"><meta property="twitter:card" content="summary"><meta property="twitter:title" content="Caldera Processor UI"><meta property="twitter:description" content="Caldera Forms processor user interface generator and components."><meta property="twitter:image" content="https://calderaforms.com/wp-content/themes/caldera-forms-dot-com/img/caldera-forms-logo.svg"></head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./" style="display: flex; align-items: center;"><img src="./image/brand_logo.svg" style="width:34px;"></a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
<a style="position:relative; top:3px;" href="https://github.com/calderawp/processor-ui/"><img width="20px" src="./image/github.png"></a></header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/CalderaProcessors.js~CalderaProcessors.html">CalderaProcessors</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-CalderaProcessorsWithState">CalderaProcessorsWithState</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-processor">components/Processor</a><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/components/Processor/Editor.js~Editor.html">Editor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/components/Processor/ProcessorLabel.js~ProcessorLabel.html">ProcessorLabel</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/components/Processor/RemoveProcessorButton.js~RemoveProcessorButton.html">RemoveProcessorButton</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/components/Processor/TypeChooser.js~TypeChooser.html">TypeChooser</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-OpenEditorButton">OpenEditorButton</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-processors">components/Processors</a><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/components/Processors/List.js~List.html">List</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-NewProcessor">NewProcessor</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#factories">factories</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-configFieldFactory">configFieldFactory</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-configFieldsFactory">configFieldsFactory</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-processorFactory">processorFactory</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-checkConfigFieldConditionals">checkConfigFieldConditionals</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-checkConfigFieldsConditionals">checkConfigFieldsConditionals</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getConfigFieldDefaultValue">getConfigFieldDefaultValue</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getConfigFieldValue">getConfigFieldValue</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getConfigFieldValueOrDefault">getConfigFieldValueOrDefault</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-mapKeysToIdProperty">mapKeysToIdProperty</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-objectToMap">objectToMap</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-reduceConfigFieldsToValues">reduceConfigFieldsToValues</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-processorTypesMap">processorTypesMap</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#processors">processors</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-emailDefaultConfigFields">emailDefaultConfigFields</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-EMAIL_PROCESSOR_TYPE">EMAIL_PROCESSOR_TYPE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-processorTypes">processorTypes</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-REDIRECT_PROCESSOR_TYPE">REDIRECT_PROCESSOR_TYPE</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#state">state</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-addProcessor">addProcessor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-newProcessor">newProcessor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-removeProcessor">removeProcessor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-setFormForProcessor">setFormForProcessor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-setProcessorType">setProcessorType</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-updateProcessor">updateProcessor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-updateProcessorConfigFields">updateProcessorConfigFields</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-updateProcessorValues">updateProcessorValues</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-clone">clone</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-processorReducer">processorReducer</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-processorTypesReducer">processorTypesReducer</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-processorsReducer">processorsReducer</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getFormForProcessor">getFormForProcessor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getProcessor">getProcessor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getProcessorFromCollection">getProcessorFromCollection</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getProcessorType">getProcessorType</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getProcessorValue">getProcessorValue</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getProcessorValues">getProcessorValues</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getProcessorsCollection">getProcessorsCollection</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ADD_PROCESSOR">ADD_PROCESSOR</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-NEW_PROCESSOR">NEW_PROCESSOR</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-REMOVE_PROCESSOR">REMOVE_PROCESSOR</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-SET_FORM_FOR_PROCESSOR">SET_FORM_FOR_PROCESSOR</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-SET_PROCESSOR_TYPE">SET_PROCESSOR_TYPE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-UPDATE_PROCESSOR">UPDATE_PROCESSOR</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-UPDATE_PROCESSOR_CONFIG_FIELDS">UPDATE_PROCESSOR_CONFIG_FIELDS</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-UPDATE_PROCESSOR_VALUES">UPDATE_PROCESSOR_VALUES</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-CALDERA_FORMS_PROCESSORS_STORE_SLUG">CALDERA_FORMS_PROCESSORS_STORE_SLUG</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-processorsStore">processorsStore</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-CALDERA_FORMS_PROCESSORS_STORE_DEFAULT_STATE">CALDERA_FORMS_PROCESSORS_STORE_DEFAULT_STATE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-CALDERA_FORMS_PROCESSOR_STORE_DEFAULT_STATE">CALDERA_FORMS_PROCESSOR_STORE_DEFAULT_STATE</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/components/Processors/List.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import React from &apos;react&apos;;
import PropTypes from &apos;prop-types&apos;;
import {RemoveProcessorButton} from &quot;../Processor/RemoveProcessorButton&quot;;
import {Editor} from &quot;../Processor/Editor&quot;;
import {OpenEditorButton} from &quot;../Processor/OpenEditorButton&quot;;
import {TypeChooser} from &quot;../Processor/TypeChooser&quot;;
import {ProcessorLabel} from &quot;../Processor/ProcessorLabel&quot;;
import {
	checkConfigFieldsConditionals,
	mapKeysToIdProperty,
	objectToMap,
	reduceConfigFieldsToValues
} from &quot;../../factories/util&quot;;

const findFieldValues = (processor) =&gt; {
	return reduceConfigFieldsToValues(
		mapKeysToIdProperty(
			objectToMap(
				processor.configFields
			)
		)
	);

}
/**
 * Displays list of processors
 *
 * @param {Object} props
 * @returns {*}
 * @constructor
 */
export class List extends React.Component{

	constructor(props){
		super(props);
		this.state = {
			opened: {},
			processorFieldValues: new Map(),
		};
		this.onUpdateProcessor = this.onUpdateProcessor.bind(this);

	}



	/**
	 * When processors is updated, do stuff
	 *
	 * @todo animations
	 * @param {Object} updatedProcessor The processor config that has been updated
	 */
	onUpdateProcessor(updatedProcessor){
		let opened = this.state.opened;
		opened[updatedProcessor.ID]=false;
		this.setState({
			opened
		});
		this.props.onUpdateProcessor(updatedProcessor);
		opened[updatedProcessor.ID]=true;

		this.setState({
			opened,
		});

		if( this.state.processorFieldValues.has( updatedProcessor.ID)){
			let previousValues = this.state.processorFieldValues.get( updatedProcessor.ID );
			const conditionalChecks = checkConfigFieldsConditionals(
				updatedProcessor.configFields,
				this.state.processorFieldValues.get( updatedProcessor.ID )
			);
			Object.keys( updatedProcessor.configFields ).forEach(configFieldId =&gt;{
				if( ! conditionalChecks[configFieldId]){
					updatedProcessor.configFields[configFieldId].value = previousValues[configFieldId];
				}
			});
		}

		this.setState({
			processorFieldValues: this.state.processorFieldValues.set(
				updatedProcessor.ID,
				findFieldValues(updatedProcessor)
			)
		});

	}

	/**
	 * Create editor for one processor
	 *
	 * Adds the conditional logic to the Editor component
	 *
	 * @param {Object} processor
	 * @param {Boolean} isOpen
	 * @return {*}
	 */
	editor(processor,isOpen){
		if( &apos;object&apos; === typeof  processor
			&amp;&amp; this.props.processors.has(processor.ID ) ){
			const conditionalChecks = checkConfigFieldsConditionals(
				processor.configFields,
				this.state.processorFieldValues.get( processor.ID )
			);
			let configFields = {};
			Object.keys( processor.configFields ).forEach(configFieldId =&gt;{
				if( conditionalChecks[configFieldId]){
					configFields[configFieldId]= processor.configFields[configFieldId];
				}
			});

			return(
				&lt;Editor
					ID={processor.ID}
					type={processor.type}
					label={processor.label}
					configFields={configFields}
					onUpdateProcessor={this.onUpdateProcessor}
					form={this.props.form}
					isOpen={isOpen}
					getProcessorFromCollection={this.props.getProcessorFromCollection}
				/&gt;
			)
		}

	}


	render() {
		const processors = Array.from( this.props.processors.values() );
		return (
			&lt;div&gt;
				{processors.map(processor =&gt; {
					let isOpen = this.state.opened.hasOwnProperty(processor.ID) &amp;&amp; true === this.state.opened[processor.ID];
					const key = processor.type ? `${processor.type}-${processor.ID}` : processor.ID;
					return (
						&lt;div
							key={key}
						&gt;
							{isOpen &amp;&amp;
								&lt;React.Fragment&gt;{this.editor(processor,isOpen)}&lt;/React.Fragment&gt;
							}

							{!isOpen &amp;&amp;
								&lt;ProcessorLabel
									ID={processor.ID}
									label={processor.label}

								/&gt;
							}

							{processor.type &amp;&amp;
								&lt;OpenEditorButton
									isOpen={isOpen}
									onClick={() =&gt;{
										let opened = this.state.opened;
										if( ! opened.hasOwnProperty(processor.ID) ){
											opened[processor.ID]=true;
										}else{
											opened[processor.ID] = !opened[processor.ID];
										}

										this.setState({
											opened
										})
									}}
								/&gt;

							}

							{! processor.type &amp;&amp;
								&lt;TypeChooser
									onUpdateProcessor={this.onUpdateProcessor}
									ID={processor.ID}
									getProcessorTypes={this.props.getProcessorTypes}
								/&gt;

							}


							&lt;RemoveProcessorButton
								onClick={() =&gt; {
									this.props.onRemoveProcessor(processor.ID);
								}}
							/&gt;
						&lt;/div&gt;
					);
				})}
			&lt;/div&gt;
		)
	}

};

List.propTypes = {
	form: PropTypes.object,
	processors: PropTypes.instanceOf(Map).isRequired,
	onRemoveProcessor: PropTypes.func.isRequired,
	onUpdateProcessor: PropTypes.func.isRequired,
	getProcessorFromCollection: PropTypes.func,
	getProcessorTypes: PropTypes.func
};
</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
