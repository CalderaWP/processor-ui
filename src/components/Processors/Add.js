import React from 'react';
import PropTypes from 'prop-types';

/**
 * Button to add new processor with
 * @param {Object} props
 * @returns {*}
 * @constructor
 */
export  const Add = (props) => {
	function handler(event){
		props.onAdd(event.target.value);
	}

	return(
		<div>
			<button
				onClick={handler}
			>
				{props.label}
			</button>
		</div>
	)
};

Add.propTypes = {
	onAdd: PropTypes.func.isRequired,
	label: PropTypes.string
};

Add.defaultProps = {
	label: 'Add Processor'
};