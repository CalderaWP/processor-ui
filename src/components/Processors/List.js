import React from 'react';
import PropTypes from 'prop-types';
import {RemoveProcessorButton} from "../Processor/RemoveProcessorButton";
import {Editor} from "../Processor/Editor";
import {OpenEditorButton} from "../Processor/OpenEditorButton";

/**
 * Displays list of processors
 *
 * @param {Object} props
 * @returns {*}
 * @constructor
 */
export class List extends React.Component{

	constructor(props){
		super(props);
		this.state = {
			opened: {}
		};
		this.onUpdateProcessor = this.onUpdateProcessor.bind(this);

	}
	onUpdateProcessor(updatedProcessor){
		let opened = this.state.opened;
		opened[updatedProcessor.ID]=false;
		this.setState({
			opened
		});
		this.props.onUpdateProcessor(updatedProcessor);
	}


	render() {
		const processors = Array.from( this.props.processors.values() );
		return (
			<div>
				{processors.map(processor => {
					let isOpen = this.state.opened.hasOwnProperty(processor.ID) && true === this.state.opened[processor.ID];
					const key = processor.type ? `${processor.type}-${processor.ID}` : processor.ID;
					return (
						<div
							key={key}
						>
							<em>{processor.ID}</em>
							{isOpen &&
								<Editor
									ID={processor.ID}
									type={processor.type}
									configFields={processor.configFields}
									onUpdateProcessor={this.onUpdateProcessor}
									form={this.props.form}
									isOpen={isOpen}
								/>
							}

							<OpenEditorButton
								isOpen={isOpen}
								onClick={() =>{
									let opened = this.state.opened;
									if( ! opened.hasOwnProperty(processor.ID) ){
										opened[processor.ID]=true;
									}else{
										opened[processor.ID] = !opened[processor.ID];
									}

									this.setState({
										opened
									})
								}}
							/>
							<RemoveProcessorButton
								onClick={() => {
									this.props.onRemoveProcessor(processor.ID);
								}}
							/>
						</div>
					);
				})}
			</div>
		)
	}

};

List.propTypes = {
	form: PropTypes.object.isRequired,
	processors: PropTypes.instanceOf(Map).isRequired,
	onRemoveProcessor: PropTypes.func.isRequired,
	onUpdateProcessor: PropTypes.func.isRequired
};
