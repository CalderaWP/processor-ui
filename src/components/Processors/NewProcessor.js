import React from 'react';
import PropTypes from 'prop-types';

/**
 * Button to add new processor with
 * @param {Object} props
 * @returns {*}
 * @constructor
 */
export  const NewProcessor = (props) => {
	function handler(event){
		props.onNewProcessor(event.target.value);
	}

	return(
		<div>
			<button
				onClick={handler}
			>
				{props.label}
			</button>
		</div>
	)
};

NewProcessor.propTypes = {
	onNewProcessor: PropTypes.func.isRequired,
	label: PropTypes.string
};

NewProcessor.defaultProps = {
	label: 'NewProcessor Processor'
};