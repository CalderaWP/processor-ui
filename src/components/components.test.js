import renderer from 'react-test-renderer';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {List} from './Processors/List';
import {NewProcessor} from './Processors/NewProcessor';

Enzyme.configure({ adapter: new Adapter() });


describe( 'Processor UI components', () => {
	const emailProcessor = {
		ID: 'p1',
		label: 'Main Email',
		type: 'email'
	};
	const processors = new Map();
	processors.set( 'p1', emailProcessor );
const form = {
	ID: 'CF1',
	fields: {

	}
}
	describe( 'Processors list component', () => {
		it( 'renders with 1 processor', () => {
			const component = renderer.create(
				<List
					form={form}
					processors={processors}
					onRemoveProcessor={() => {}}
					onUpdateProcessor={() => {}}
				/>
			);
			expect( component.toJSON( ) ).toMatchSnapshot();
		});

		it( 'renders with 2 processors', () => {
			const twoProcessors = new Map();
			twoProcessors.set( 'p1', emailProcessor );
			twoProcessors.set( 'p2', {
				ID: 'p21',
				label: 'Secondary Email',
				type: 'email'
			} );

			const component = renderer.create(
				<List
					form={form}
					processors={processors}
					onRemoveProcessor={() => {}}
					onUpdateProcessor={() => {}}
				/>
			);
			expect( component.toJSON( ) ).toMatchSnapshot();
		});

	});

	describe( 'Processor add button', () => {
		describe( 'Renders with default label', () => {
			it( 'matches snapshot', () => {
				const component = renderer.create(
					<NewProcessor onNewProcessor={()=>{}}/>
				);
				expect( component.toJSON( ) ).toMatchSnapshot();
			});

			it( 'Has the default label', () => {
				const wrapper = shallow(
					<NewProcessor onNewProcessor={()=>{}}/>
				);
				expect( wrapper.text( ) ).toEqual( NewProcessor.defaultProps.label );
			});

		});

		describe( 'Renders with  label prop', () => {
			it( 'matches snapshot', () => {
				const component = renderer.create(
					<NewProcessor
						onNewProcessor={()=>{}}
						label={'Hi Roy'}
					/>
				);
				expect( component.toJSON( ) ).toMatchSnapshot();
			});

			it( 'Has the default label', () => {
				const wrapper = shallow(
					<NewProcessor
						onNewProcessor={()=>{}}
						label={'Hi Roy'}
					/>
				);
				expect( wrapper.text( ) ).toEqual( 'Hi Roy' );
			});

		});

	});


});