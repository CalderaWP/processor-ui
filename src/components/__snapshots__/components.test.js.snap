// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processor UI components Processor add button Renders with  label prop matches snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Hi Roy
  </button>
</div>
`;

exports[`Processor UI components Processor add button Renders with default label matches snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    NewProcessor Processor
  </button>
</div>
`;

exports[`Processor UI components Processors list component renders with 1 processor 1`] = `
<div>
  <div>
    <em>
      EmailProcessorId
    </em>
    <button
      className={undefined}
      onClick={[Function]}
      style={undefined}
    >
      <span>
        Edit Settings
      </span>
    </button>
    <button
      className={undefined}
      onClick={[Function]}
      style={undefined}
    >
      <span>
        Remove Processor
      </span>
    </button>
  </div>
</div>
`;

exports[`Processor UI components Processors list component renders with 2 processors 1`] = `
<div>
  <div>
    <em>
      EmailProcessorId
    </em>
    <button
      className={undefined}
      onClick={[Function]}
      style={undefined}
    >
      <span>
        Edit Settings
      </span>
    </button>
    <button
      className={undefined}
      onClick={[Function]}
      style={undefined}
    >
      <span>
        Remove Processor
      </span>
    </button>
  </div>
</div>
`;
